.github/actions/setup_xcode_build_cache/action.yml

```diff
@@ -0,0 +1,28 @@
+name: setup_xcode_build_cache
+description: Add caching to iOS jobs to speed up builds
+inputs:
+  hermes-version:
+    description: The version of hermes
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: See commands.yml with_xcodebuild_cache
+      shell: bash
+      run: echo "See commands.yml with_xcodebuild_cache"
+    - name: Prepare Xcodebuild cache
+      shell: bash
+      run: |
+        WEEK=$(date +"%U")
+        YEAR=$(date +"%Y")
+        echo "$WEEK-$YEAR" > /tmp/week_year
+    - name: Cache podfile lock
+      uses: actions/cache@v4.0.0
+      with:
+        path: packages/rn-tester/Podfile.lock
+        key: v9-podfilelock-${{ github.job }}-${{ hashfiles('packages/rn-tester/Podfile') }}-{{ hashfiles('/tmp/week_year') }}-${{ inputs.hermes-version}}
```

Comment:
I have seen folks use a date, or part of a date, in their cache key. I think this is a reasonable thing to do, if that aligns with your cache strategy. In this case, you are probably in a better position than I do decide if this is appropriate here.  

I would probably exclude this from the cache key. I would only add this if I find it is necessary.  

That said, what strikes me as a little odd here, is that we are writing this info to a file, then hashing the file. Why not just use the values in the cache key itself, rather than writing to the file?

Reply:
The reason why we are writing it to a file is a limitation of CircleCi, which doesn't allow calls or dynamic variables inside the key. That's why a common approach is to write a variable on a file and then to hash the file in the key.

If GitHub doesn't have the limitation, I'm all in for simplifying it!

Reply:
Looking into this further, we would need to set outputs for these values, then use those outputs. Not sure if these would need to be in different steps or not. Since this is only used once, it's probably best to leave this as is, unless you want to see the values in the cache key rather than than hash.